ydb/services/metadata/manager/generic_manager.h

```diff
@@ -42,35 +44,123 @@ class TGenericOperationsManager: public IObjectOperationsManager<T> {
                     break;
-                case IOperationsManager::EActivityType::Alter:
+                case EActivityType::Alter:
                     modifyObjectCommand = std::make_shared<TUpdateObjectCommand<T>>(patch.GetResult(), manager, controller, context);
                     break;
-                case IOperationsManager::EActivityType::Drop:
-                    modifyObjectCommand = std::make_shared<TDeleteObjectCommand<T>>(patch.GetResult(), manager, controller, context);
+                case EActivityType::Drop:
+                    modifyObjectCommand = std::make_shared<TDeleteObjectCommand<T>>(patch.GetResult(), manager, controller, context, settings.GetMissingOk());
                     break;
-                case IOperationsManager::EActivityType::Undefined:
+                case EActivityType::Undefined:
                     return NThreading::MakeFuture<TYqlConclusionStatus>(TYqlConclusionStatus::Fail("undefined action type"));
             }
             TActivationContext::Send(new IEventHandle(NProvider::MakeServiceId(nodeId), {}, new NProvider::TEvObjectsOperation(modifyObjectCommand)));
         }
+        return promise;
+    }
+
+    virtual TYqlConclusionStatus DoPrepare(NKqpProto::TKqpSchemeOperation& schemeOperation, const NYql::TObjectSettingsImpl& settings,
+        const IClassBehaviour::TPtr& manager, TInternalModificationContext& context) const override {
+        if (!manager) {
+            return TYqlConclusionStatus::Fail("modification object behaviour not initialized");
+        }
+        if (!manager->GetOperationsManager()) {
+            return TYqlConclusionStatus::Fail("modification is unavailable for " + manager->GetTypeId());
+        }
+
+        TOperationParsingResult patch(TBase::BuildPatchFromSettings(settings, context));
+        if (!patch.IsSuccess()) {
+            return TYqlConclusionStatus::Fail(patch.GetErrorMessage());
+        }
+
+        NKqpProto::TKqpPhyMetadataOperation* operationProto = nullptr;
+        switch (context.GetActivityType()) {
+            case EActivityType::Upsert:
+                operationProto = schemeOperation.MutableUpsertObject();
+                break;
+            case EActivityType::Create:
+                operationProto = schemeOperation.MutableCreateObject();
+                operationProto->SetSuccessOnAlreadyExists(settings.GetExistingOk());
+                break;
+            case EActivityType::Alter:
+                operationProto = schemeOperation.MutableAlterObject();
+                break;
+            case EActivityType::Drop:
+                operationProto = schemeOperation.MutableDropObject();
+                operationProto->SetSuccessOnNotExist(settings.GetMissingOk());
+                break;
+            case EActivityType::Undefined:
+                return TYqlConclusionStatus::Fail("undefined action type");
+        }
+        Y_DEBUG_ABORT_UNLESS(operationProto);
```

Comment:
We return Success in release mode, is that what we want? Maybe we should use Y_ENSURE here and abort the request?

Reply:
We don't return Success here: we will simply segfault because we use operationProto :)
I think this is an internal logical error, because all options for ActivityType are listed here. Futher more, the compiler would tell us if additional option exists.

Reply:
I would prefer Y_ENSURE to get an exception in release instead of a segfault nonetheless